library/std/src/sys/thread_local/os.rs

```diff
@@ -31,13 +32,76 @@ pub macro thread_local_inner {
             $crate::thread::LocalKey::new(|init| {
                 static VAL: $crate::thread::local_impl::Storage<$t>
                     = $crate::thread::local_impl::Storage::new();
-                VAL.get(init, __init)
+                VAL.get($crate::thread::local_impl::thread_local_inner!(@align $(#[$($align_attr)*])*), init, __init)
             })
         }
     }},
-    ($(#[$attr:meta])* $vis:vis $name:ident, $t:ty, $($init:tt)*) => {
+
+    // Handle `rustc_align_static` attributes,
+    // by translating them into an argumemt to pass to `Storage::get`:
+
+    // fast path for when there are none
+    (@align) => (1),
+
+    // `rustc_align_static` attributes are present,
+    // translate them into a `const` block that computes the alignment
```

Comment:
I wonder if this could be simplified by attaching the attributes to some static and then checking the alignment of that static (via align_of_val)? That way we presumably get most of the error checking from rustc and we don't nee to manually compute the alignment or deal with cfgs.

Reply:
Perhaps this was meant as a comment on the struct idea below? For this comment using a static shouldn't require any translation - presumably align_of_val(&DUMMY) gives us the alignment we need?